File: HashSort/HashSort.cs
using System;

class Program
{

    static void randomnumber(int[] a)
    {
        Random rand = new Random();
        for (int i = 0; i < a.Length; i++)
        {
            a[i] = rand.Next(1, 100);
        }
    }

    static void printarray(int[] a)
    {
        for (int i = 0; i < a.Length; i++)
        {
            Console.Write(a[i] + " ");
        }
        Console.WriteLine();
    }

    static void shellSort(int[] arr)
    {
        int n = arr.Length;
        for (int gap = n / 2; gap > 0; gap /= 2)
        {
            for (int i = gap; i < n; i++)
            {
                int temp = arr[i];
                int j;
                for (j = i; j >= gap && arr[j - gap] > temp; j -= gap)
                {
                    arr[j] = arr[j - gap];
                }
                arr[j] = temp;
            }
        }
    }

    static void Main()
    {
        int[] arr = new int[15];
        int longitud = arr.Length;
        randomnumber(arr);
        Console.WriteLine("Array Original: ");
        printarray(arr);
        shellSort(arr);
        Console.WriteLine("Array Ordenado con Heap Sort: ");
        printarray(arr);
    }
}

File: HashSort/HashSort.py
import random

def randomnumber(a):
    for i in range(len(a)):
        a[i] = random.randint(1, 100)

def printarray(a):
    for i in range(len(a)):
        print(a[i], end=" ")
    print()


def ShellSort(a):
    n = len(a)
    gap = n // 2
    while gap > 0:
        for i in range(gap, n):
            temp = a[i]
            j = i
            while j >= gap and a[j - gap] > temp:
                a[j] = a[j - gap]
                j -= gap
            a[j] = temp
        gap //= 2
    return a


array = [0] * 15
randomnumber(array) 
longitud = len(array)
print("Arreglo Original: ")
printarray(array)
ShellSort(array)  
print("Array Ordenado:")
printarray(array)        
